go/vt/vttablet/tabletserver/connpool/dbconn.go

```diff
@@ -163,17 +163,38 @@ func (dbc *Conn) execOnce(ctx context.Context, query string, maxrows int, wantfi
 
 	defer dbc.stats.MySQLTimings.Record("Exec", time.Now())
 
-	done, wg := dbc.setDeadline(ctx)
-	qr, err := dbc.conn.ExecuteFetch(query, maxrows, wantfields)
+	resultChan := make(chan *sqltypes.Result, 1)
+	errChan := make(chan error, 1)
 
-	if done != nil {
-		close(done)
-		wg.Wait()
+	startTime := time.Now()
```

Comment:
making channel on runtime for every call seems expensive to me.
I have added `Benchmark me` label to it as well.

Reply:
This is no worse than the current code, which already creates a channel _and_ a waitgroup. The current code, however, is very wasteful. So I think we can improve this to make this even less costly! Let me try...